ibtk/src/lagrangian/FEProjector.cpp

```diff
@@ -389,40 +443,29 @@ FEProjector::computeL2Projection(PetscVector<double>& U_vec,
     bool converged = false;
 
     if (close_F) F_vec.close();
-    const System& system = d_fe_data->getEquationSystems()->get_system(system_name);
-    const DofMap& dof_map = system.get_dof_map();
-    if (consistent_mass_matrix)
-    {
-        std::pair<PetscLinearSolver<double>*, PetscMatrix<double>*> proj_solver_components =
-            buildL2ProjectionSolver(system_name);
-        PetscLinearSolver<double>* solver = proj_solver_components.first;
-        PetscMatrix<double>* M_mat = proj_solver_components.second;
-        PetscBool rtol_set;
-        double runtime_rtol;
-        ierr = PetscOptionsGetReal(nullptr, "", "-ksp_rtol", &runtime_rtol, &rtol_set);
-        IBTK_CHKERRQ(ierr);
-        PetscBool max_it_set;
-        int runtime_max_it;
-        ierr = PetscOptionsGetInt(nullptr, "", "-ksp_max_it", &runtime_max_it, &max_it_set);
-        IBTK_CHKERRQ(ierr);
-        ierr = KSPSetFromOptions(solver->ksp());
-        IBTK_CHKERRQ(ierr);
-        solver->solve(
-            *M_mat, *M_mat, U_vec, F_vec, rtol_set ? runtime_rtol : tol, max_it_set ? runtime_max_it : max_its);
-        KSPConvergedReason reason;
-        ierr = KSPGetConvergedReason(solver->ksp(), &reason);
-        IBTK_CHKERRQ(ierr);
-        converged = reason > 0;
-    }
-    else
-    {
-        PetscVector<double>* M_diag_vec = buildDiagonalL2MassMatrix(system_name);
-        ierr = VecPointwiseDivide(U_vec.vec(), F_vec.vec(), M_diag_vec->vec());
-        IBTK_CHKERRQ(ierr);
-        converged = true;
-    }
+    std::pair<PetscLinearSolver<double>*, PetscMatrix<double>*> proj_solver_components =
```

Comment:
Can we just do a pointwise divide in cases where the lumped mass matrix remains diagonal?

Reply:
That should be possible (or a pointwise mult). I'll take a look.

Reply:
Detecting whether or not we have constraints or something else that causes a non-diagonal lumped mass matrix is tricky with the way this is currently set up. I've already spent half an hour on this and I suspect its going to take me at least another hour to finish this for virtually no benefit (converging in one iteration is approximately the same amount of work). Can we skip this?

Reply:
Can you quickly confirm that there is negligible runtime performance impact for problems without constraints?

Reply:
Sure!

Another thought I had: if we ever do time-dependent FE AMR + nodal quadrature then this check will get even more complicated.

Reply:
It looks like the linear solve takes up about 5% of the total run time of `interpolateVelocity`: it's low but since its measurable I suppose we should try to skip the solve.

Reply:
Fixed!